src/__test__/components/plots-and-tables/markerHeatmap.test.jsx

```diff
@@ -217,48 +217,48 @@ describe('Marker heatmap plot', () => {
   it('loads marker genes on selecting', async () => {
     await renderHeatmapPage(defaultStore);
 
-    const geneSelection = rtl.screen.getByText('Gene selection');
-    userEvent.click(geneSelection);
-    const nGenesInput = rtl.getByTestId(geneSelection.parentElement, 'num-genes');
+    const markerGenes = screen.getByText('Marker genes');
+
+    userEvent.click(markerGenes);
+    const nGenesInput = screen.getByRole('spinbutton', { name: 'Number of genes input' });
     userEvent.type(nGenesInput, 10);
 
-    const runButton = rtl.getByText(geneSelection.parentElement, 'Run');
-    userEvent.click(runButton);
+    userEvent.click(screen.getByText('Run'));
 
-    await rtl.waitFor(() => expect(loadMarkersSpy).toHaveBeenCalled());
+    await waitFor(() => expect(loadMarkersSpy).toHaveBeenCalled());
```

Comment:
I think this check could be expanded somehow. I can imagine very easily situations where the users selects 10 genes and the call is actually done (so the test won't fail) but the actual call or the result is wrong.